client/src/redux/theme-saga.js

```diff
@@ -0,0 +1,42 @@
+import { put, takeEvery, select, take } from 'redux-saga/effects';
+import { createFlashMessage } from '../components/Flash/redux';
+import { setTheme } from './actions';
+import { actionTypes } from './action-types';
+import { userThemeSelector } from './selectors';
+
+function* toggleThemeSaga() {
+  const data = { type: 'success', message: 'flash.updated-themes' };
+  const currentTheme = localStorage.getItem('theme');
+  const invertedTheme = currentTheme === 'dark' ? 'light' : 'dark';
+  localStorage.setItem('theme', invertedTheme);
+  yield put(setTheme(invertedTheme));
+  yield put(createFlashMessage({ ...data }));
+}
+
+function* initalizeThemeSaga() {
+  // Wait for the fetch userComplete action
+  yield take(actionTypes.fetchUserComplete);
+
+  let selectTheme;
+  const isUserThemeDark = (yield select(userThemeSelector)) === 'night';
```

Comment:
Is the idea that we're using the db theme initially, so we don't overwrite people's settings? If so, a comment explaining that this is a temporary measure could be useful.

Reply:
Yes, once the local storage key is set db theme is ignored. I will expand the comment below.